rclpy/rclpy/node.py

```diff
@@ -83,6 +84,10 @@
 SrvTypeRequest = TypeVar('SrvTypeRequest')
 SrvTypeResponse = TypeVar('SrvTypeResponse')
 
+# Re-export exception defined in _rclpy C extension.
+# `Node.get_*_names_and_types_by_node` methods may raise this error.
+NodeNameNonExistentError = _rclpy.NodeNameNonExistentError
```

Comment:
@ivanpauno there was [a previous comment](https://github.com/ros2/rclpy/pull/413#discussion_r319660573) from @dirk-thomas about not exporting to Python exceptions defined in C but the other way around. Since modules like `rclpy.qos` already import C extensions at the module level, this does not worsen the situation in any way. 

So I'd say that we merge this as-is, to solve https://github.com/ros2/build_cop/issues/223, and then we work towards normalizing C extension use in `rclpy` -- whatever that means. There's a PR, https://github.com/ros2/rclpy/pull/422, for it already. @dirk-thomas do you agree?